src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -474,19 +473,37 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
 		_mavlink->configure_stream_threadsafe("HOME_POSITION", 0.5f);
 
 	} else if (cmd_mavlink.command == MAV_CMD_SET_MESSAGE_INTERVAL) {
-		if (set_message_interval((int)(cmd_mavlink.param1 + 0.5f), cmd_mavlink.param2, cmd_mavlink.param3)) {
+		if (set_message_interval((int)std::round(cmd_mavlink.param1), cmd_mavlink.param2, cmd_mavlink.param3)) {
 			result = vehicle_command_ack_s::VEHICLE_RESULT_FAILED;
 		}
 
 	} else if (cmd_mavlink.command == MAV_CMD_GET_MESSAGE_INTERVAL) {
-		get_message_interval((int)cmd_mavlink.param1);
+		get_message_interval((int)std::round(cmd_mavlink.param1));
 
 	} else if (cmd_mavlink.command == MAV_CMD_REQUEST_FLIGHT_INFORMATION) {
 		send_flight_information();
 
 	} else if (cmd_mavlink.command == MAV_CMD_REQUEST_STORAGE_INFORMATION) {
-		if ((int)(cmd_mavlink.param2 + 0.5f) == 1) {
-			send_storage_information(cmd_mavlink.param1 + 0.5f);
+		if ((int)std::round(cmd_mavlink.param1) == 1) {
+			send_storage_information((int)std::round(cmd_mavlink.param1));
+		}
+
+	} else if (cmd_mavlink.command == MAV_CMD_REQUEST_MESSAGE) {
+		uint16_t message_id = (uint16_t)std::round(cmd_mavlink.param1);
+		bool stream_found = false;
+
+		for (const auto &stream : _mavlink->get_streams()) {
+			if (stream->get_id() == message_id) {
+				stream->request_message(vehicle_command.param1, vehicle_command.param2, vehicle_command.param3,
```

Comment:
Why are all the parameters passed on?

Reply:
For my use case (see below), I need parameters 2, 3, and 4 (although the specifications for this are still under discussion, so that might not be the case in the end). Based on this, it seems to be fine to use any set of the parameters. So for future implementations of this method, different messages might have to use any or all of the parameters from the command.

Reply:
I think a good idea. We could just update the mavlink message to define all these param values properly now if that will make it easy.

Reply:
That's indeed a bit odd. According to the [message definition](https://mavlink.io/en/messages/common.html#MAV_CMD_REQUEST_MESSAGE) no params should be really used, right?

Reply:
What @ItsTimmy has done is the intention of the message, but not its current implementation. 

Specifically, some message, like the proposed microservices requestor, depend on parameters. If we want to use this message to request it we need to be able to provide those parameters. So if we don't do this, we'd need a dedicated message (also a valid option).

I've updated the message so that it is more clear: https://github.com/mavlink/mavlink/pull/1276

Reply:
@hamishwillee ok got it. I think in that case I just don't like the fact that we overload the REQUEST_MESSAGE with additional params. I can see how we got there but it makes it hard to understand from the outside, in my opinion.

Reply:
@julianoes That's a reasonable argument. Certainly having dedicated messages makes microservices more compartmentalised.

But it does mean we need to create at least one dedicated message for @ItsTimmy to use for this. Tim, can you propose a message in the spec?